megalinter/Linter.py

```diff
@@ -1076,6 +1090,43 @@ def get_total_number_errors(self, stdout: str):
         else:
             return 1
 
+    # Find number of warnings in linter stdout log
+    def get_total_number_warnings(self, stdout: str):
+        total_warnings = 0
+
+        # Get number with a single regex.
+        if self.cli_lint_warnings_count == "regex_number":
+            reg = self.get_regex(self.cli_lint_warnings_regex)
+            m = re.search(reg, stdout)
+            if m:
+                total_warnings = int(m.group(1))
+        # Count the number of occurrences of a regex corresponding to an error in linter log (parses linter log)
+        elif self.cli_lint_warnings_count == "regex_count":
+            reg = self.get_regex(self.cli_lint_warnings_regex)
+            total_warnings = len(re.findall(reg, stdout))
+        # Sum of all numbers found in linter logs with a regex. Found when each file prints out total number of errors
+        elif self.cli_lint_warnings_count == "regex_sum":
+            reg = self.get_regex(self.cli_lint_warnings_regex)
+            matches = re.findall(reg, stdout)
+            total_warnings = sum(int(m) for m in matches)
+        # Count all lines of the linter log
+        elif self.cli_lint_warnings_count == "total_lines":
+            total_warnings = sum(
+                not line.isspace() and line != "" for line in stdout.splitlines()
+            )
+        # Return result if found, else default value according to status
+        if total_warnings > 0:
+            return total_warnings
+        if self.cli_lint_warnings_count is not None:
```

Comment:
it will display a warning in log for each linter which does not have cli_lint_warnings_count
Please could you let None value, then display nothing in the warning column instead of 0 ?

Reply:
I believe the change I just made should fix this as warnings would display one sometimes based on status